src/pages/home/sidebar/SidebarLinksData.js

```diff
@@ -63,24 +68,26 @@ const defaultProps = {
     policies: {},
 };
 
-function SidebarLinksData({isFocused, allReportActions, betas, chatReports, currentReportID, insets, isLoadingReportData, onLinkClick, policies, priorityMode}) {
+function SidebarLinksData({isFocused, allReportActions, betas, chatReports, currentReportID, insets, isLoadingReportData, onLinkClick, policies, priorityMode, network}) {
     const styles = useThemeStyles();
     const {translate} = useLocalize();
 
     const reportIDsRef = useRef(null);
     const isLoading = SessionUtils.didUserLogInDuringSession() && isLoadingReportData;
     const optionListItems = useMemo(() => {
         const reportIDs = SidebarUtils.getOrderedReportIDs(null, chatReports, betas, policies, priorityMode, allReportActions);
+
         if (deepEqual(reportIDsRef.current, reportIDs)) {
             return reportIDsRef.current;
         }
 
         // We need to update existing reports only once while loading because they are updated several times during loading and causes this regression: https://github.com/Expensify/App/issues/24596#issuecomment-1681679531
-        if (!isLoading || !reportIDsRef.current) {
+        // However, if the user is offline, we need to update the reports unconditionally, since the loading of report data might be stuck in this case.
+        if (!isLoading || !reportIDsRef.current || network.isOffline) {
             reportIDsRef.current = reportIDs;
         }
         return reportIDsRef.current || [];
-    }, [allReportActions, betas, chatReports, policies, priorityMode, isLoading]);
+    }, [allReportActions, betas, chatReports, policies, priorityMode, isLoading, network]);
```

Comment:
can we specify directly network.isOffline in deps and not the whole network object?

Reply:
Yes, that is what I originally tested with. That also works perfectly.

I changed it to be the whole object instead because it visually looked neater, and I didn't find that object has any other attributes apart from isOffline that change quite frequently.

Would you like me to instead commit the version where network is replaced with network.isOffline?